### ISSUE TITLE
Use Vim8.1 popup feature

### ISSUE THREAD

**Issue by rhysd** [rhysd/git-messenger.vim#30]
https://github.com/vim/vim/blob/master/runtime/doc/popup.txt

It seems about to be being implemented. If it works and sufficient for this plugin, using it would be better for Vim users.

**Comment by airblade** [comment]
As far as I understand a popup window cannot be focused so it wouldn't be possible to use the mappings etc.

**Comment by rhysd** [comment]
Yeah, the point is a blocker. Currently I don't have good idea to workaround that.

**Comment by airblade** [comment]
I think if the user wants to use popups to see the blame info, it's useful even if they can't do anything with it.  I suppose you can make the popup as big as the blame info, up to the size of the window, to minimise scrolling.

**Comment by rhysd** [comment]
Current implementation determines size of float window following contents. However, sometimes commit body is too big. And if user wants to see diff of the commit, it would be not fit to window (diff is usually bigger).

Another point is local key maps in the float window. This plugin provides some mappings such as `o` (show older commit). I also don't have good idea to provide the mappings without focusing (entering) a popup. Neovim's float window allows to move cursor into popup, so we can define window-local mappings. But that seems not available in Vim.

**Comment by blayz3r** [comment]
https://github.com/vim/vim/issues/4577

**Comment by tracyone** [comment]
```vim
:help previewpopup
```

**Comment by blayz3r** [comment]
bump

**Comment by tsuyoshicho** [comment]
I think operation likely that.
[bfrg/vim\-qf\-preview: Vim plugin to preview current quickfix item in a popup window \(Vim\-only\)](https://github.com/bfrg/vim-qf-preview)

(skip into preview; direct operation.)

**Comment by sheerun** [comment]
Also the window currently opened  immediately closes... I don't know why

**Comment by rhysd** [comment]
@sheerun That's not related to this issue. See https://github.com/rhysd/git-messenger.vim/issues/54#issuecomment-667542845

**Comment by blayz3r** [comment]
```vim
nmap <silent><Leader>g :call setbufvar(winbufnr(popup_atcursor(split(system("git log -n 1 -L " . line(".") . ",+1:" . expand("%:p")), "\n"), { "padding": [1,1,1,1], "pos": "botleft", "wrap": 0 })), "&filetype", "git")<CR>
```
![image](https://user-images.githubusercontent.com/12205650/89588928-a3f56200-d812-11ea-816c-1887b1d72804.png)

co -> https://www.reddit.com/user/sheerun/

**Comment by nickjj** [comment]
Are Vim 8.1 and 8.2's popup windows the same?

Just asking because with Vim 8.2 along with FZF you can set `let g:fzf_layout = { 'window': { 'width': 0.5, 'height': 0.6 } }` which gives you a floating window style popup that you can interact with and move the cursor in. Even mouse events work.

There's lots of discussion around it and Vim script snippets at https://github.com/junegunn/fzf.vim/issues/821. That's in the context of FZF of course, but maybe that method could be applied here too?

**Comment by rhysd** [comment]
All right, I'll make time to try to implement this feature this week.

**Comment by blayz3r** [comment]
Discussion here:
https://github.com/vim/vim/issues/5639#issuecomment-593419210

**Comment by mmrwoods** [comment]
Hi, I've added a PR to add support for Vim 8.1+ popup windows here: https://github.com/rhysd/git-messenger.vim/pull/99

This is working nicely for me, I'd appreciate any feedback, I'm sure there will be bugs found and improvements to be made!
